### ISSUE TITLE
implicit override not reported for overloaded methods without implementation

### ISSUE THREAD

**Issue by jorenham** [microsoft/pyright#10645]
Overloaded methods without implementation (i.e. in a `Protocol` or a `.pyi`) are not flagged when using `reportImplicitOverride=true` if they override a method but not decorated as such:

```py
from typing import Literal, Protocol, overload

class BoolFormatter(Protocol):
    def format_bool(self, b: bool, /) -> str: ...

class LiteralBoolFormatter(BoolFormatter, Protocol):
    @overload
    def format_bool(self, b: Literal[True], /) -> Literal["⊤"]: ...
    @overload
    def format_bool(self, b: Literal[False], /) -> Literal["⊥"]: ...
    @overload  # I'll open another issue for this 
    def format_bool(self, b: bool, /) -> Literal["⊤", "⊥"]: ...
```

[playground](https://pyright-play.net/?pythonVersion=3.14&strict=true&reportImplicitOverride=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFAArgxgDGYNUYAbmZWOQCYAoQc0rkAzuKgAhMGwBiuCORikQACgZgmrSgEoAXIKgmo-EsCjAlKgPoAjOZXXiSlYLXsGojtrQD0elAAtAB8UOIwIN4AdHHCohJSRGpUsgo2qmTq6ZSKIMpZILRaOmyGxqYAAlw8fEKmZhZWmQ5OLm4ePt4pZFQA2gAqIACuJAC6AUFhhMR9lP0ARIAlRIvjsfGNNdwgvAKVJuaW1gV2vs6u7p49cxQL8lSuk1CBIeG9d0uApURrGzEHUG2dQEJgAxFAAJIAckolA4CBIKCg5BQ2gAFmRMJIxi0QLA0UgpACjrjCm02B0rt0fE4pm9ZqkFitFrRFj91lA4jEgA)

In a `.pyi` this false negative also occurs outside of a `Protocol`:

```pyi
from typing import Literal, overload

class BaseFormatter:
    def format_bool(self, b: bool, /) -> str: ...

class Formatter(BaseFormatter):
    @overload
    def format_bool(self, b: Literal[True], /) -> Literal["⊤"]: ...
    @overload
    def format_bool(self, b: Literal[False], /) -> Literal["⊥"]: ...
    @overload
    def format_bool(self, b: bool, /) -> Literal["⊤", "⊥"]: ...
```

**Comment by erictraut** [comment]
Yeah, I agree this is inconsistent. I suspect it's related to [this issue](https://github.com/microsoft/pyright/issues/6158).
